### ISSUE TITLE
Code style

### ISSUE THREAD

**Issue by ePirat** [pupnp/pupnp#151]
Currently it is quite a wild mix of code styles, even mix between tabs and spaces so I am wondering what is what should be used?

- Indent using tabs or spaces?
- Indentation depth? (2, 4, 8…)
- Spaces around conditions or not? `if ( cond )` vs `if (cond)`
- Spaces in function calls or not? `func( arg );` vs `func(arg);` or even `func ( arg );`
- Opening brace after if or on newline?

**Comment by ePirat** [comment]
Follow-up question would be naming conventions for new headers, currently its a mix there too:

```
ActionComplete.h
EventSubscribe.h
```
vs.
```
ithread.h
upnpconfig.h
```

**Comment by mrjimenez** [comment]
Generally speaking, coding style is an inconsistent mess in this project. I am trying to put things uniform by using clang-format. There is a configuration file in the root of the project now, we could try to use it so we only have to worry in the corner cases.

I am really not radical about it, because it is impossible to keep absolute coding style consistency. A code formatter should make this task easier, but it's not like I will reject a patch because of style.

In the case of upnpconfig.h, the real file to look is upnpconfig.h.in.

I think the current style uses tabs for the first part and the tab size and indentation is 8. The rest is automatic, seems based on the kernel coding style.

```
int f(arg)
{
        if (cond) {
                func1(arg);
        } else {
                func2(arg);
        }

        return 0;
}

```

**Comment by Vollstrecker** [comment]
Closed by f2f32949325972cb16
